source/include/localisation/cs_TrackingFilter.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Mar 3, 2021
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <third/cuckoo/CuckooFilter.h>
+
+
+// -------------------------------------------------------------
+// --------------- Filter definitions and storage --------------
+// -------------------------------------------------------------
+
+/**
+ * Determines whether the filter has mac or advertisement data as input.
+ */
+enum class FilterInputType : uint8_t {
+	MacAddress = 0,
+	AdData = 1,
+};
+
+/**
+ * Data associated to a parsing filter that is persisted to flash.
+ */
+struct __attribute__((__packed__)) TrackingFilterMetaData {
+public:
+	// sync info
+	uint8_t protocol; // determines implementation type of filter
+	union {
+		struct {
+			uint16_t value: 12; // Lollipop
+			uint16_t unused : 4; // explicit padd
+		} version;
+		uint16_t version_int;
+	};
+
+	uint8_t profileId; // devices passing the filter are assigned this profile id
+	FilterInputType inputType;
+
+	// sync state
+	union{
+		uint8_t flags;
+		struct {
+			uint8_t isActive : 1;
+		} bits;
+	};
+};
+
+/**
+ * Data associated to a parsing filter that is not persisted to flash.
+ */
+struct __attribute__((__packed__)) TrackingFilterRuntimeData {
+	uint8_t filterId; 				// can be runtime only since it's saved as part of record metadata on flash.
+	uint16_t crc;
+};
+
+/**
+ * The filters and their associated metadata.
+ * Allocated on the FilterBuffer are packed into a single
+ * struct. That saves a bit of administration.
+ *
+ * Warning: keep this structure and order as-is, that makes chunking
+ * the whole thing a stupid load simpler.
+ */
+struct __attribute__((__packed__)) TrackingFilter {
```

Comment:
Not sure which struct you meant here:

Since we construct the filter contents on the host, and this data is sent over the air, both `TrackingFilterMetaData` and `CuckooFilter` must be packed. Packing the `TrackingfilterRuntimeData` as well ensures that the complete representation in the `_filterBuffer` is as small as possible and that we will not have to deal with unexpected padding issues. The (packed) wrapper object `TrackingFilter` is mostly just a practicality.